TeamCode/src/main/java/org/firstinspires/ftc/teamcode/subsystems/stateMachineController/StateMachine.java

```diff
@@ -0,0 +1,37 @@
+package org.firstinspires.ftc.teamcode.subsystems.stateMachineController;
+
+import java.util.HashMap;
+import java.util.ArrayList;
+
+public class StateMachine {
+    /* Storing states in <Class, Instance of Class> pairs.*/
+    HashMap<Class <? extends State>, State> states;
+    State currentState;
```

Comment:
You're never actually setting this to anything other than `null`, so you'll always get the `Current state is null. Check if initial state was set.` exception.